wideq/client.py

```diff
@@ -415,41 +451,94 @@ class Device(object):
         """Create a wrapper for a `DeviceInfo` object associated with a
         `Client`.
-        """
+        """        
         self.client = client
         self.device = device
         self.model: ModelInfo = client.model_info(device)
 
+    def _set_operation(self, value):
+        """Set a device's operation for a given `value`.
+        """
+        
+        self.client.session.set_device_controls(
+            self.device.id,
+            value,
+            )
+
     def _set_control(self, key, value):
         """Set a device's control for `key` to `value`."""
         self.client.session.set_device_controls(
             self.device.id,
             {key: value},
-        )
+            )
 
     def _get_config(self, key):
         """Look up a device's configuration for a given value.
-
         The response is parsed as base64-encoded JSON.
         """
         data = self.client.session.get_device_config(
-            self.device.id,
-            key,
+               self.device.id,
+               key,
         )
         return json.loads(base64.b64decode(data).decode('utf8'))
-
+    
     def _get_control(self, key):
         """Look up a device's control value."""
         data = self.client.session.get_device_config(
-            self.device.id,
-            key,
-            'Control',
+               self.device.id,
+                key,
+               'Control',
         )
 
-        # The response comes in a funky key/value format: "(key:value)".
+            # The response comes in a funky key/value format: "(key:value)".
         _, value = data[1:-1].split(':')
         return value
 
+    def _set_control_ac_wdirstep(self, key, value):
+        """set ac wind dir step.
+        """
+        
+        self.client.session.set_device_controls(
+            self.device.id,
+            {key: value},
+            )
+
+    def _delete_permission(self):
+        """for use mobile app and HA component"""
+        self.client.session.delete_permission(
+            self.device.id,
+        )
+    
+        
+    def _get_power_data(self, sDate, eDate):
+        """ get AC Power Consumption Data"""
+        period = 'Day_'+sDate+'T000000Z/'+eDate+'T000000Z'
+        data = self.client.session.get_power_data(
+               self.device.id,
+                period,
+        )
+        return data
+
+    def _get_water_usage(self, typeCode, sDate, eDate):
+        """ get water purifier using data"""
+        data = self.client.session.get_water_usage(
+               self.device.id,
+                typeCode,
+                sDate,
+                eDate,
+        )
+        return data
+
+    def _get_dustsensor_data(self):
+        """get dustsensor data"""
+        data = self.client.session.get_dustsensor_data()
+        code = res.get('returnCd')
+        if code != '0000':
+            raise MonitorError(device_id, code)
+        else:
+           return data
```

Comment:
These methods seem to be AC-specific. Maybe they should be located in the `ac` module?